pkg/controller/volume/attachdetach/attach_detach_controller.go

```diff
@@ -325,8 +339,9 @@ func (adc *attachDetachController) Run(stopCh <-chan struct{}) {
 	adc.broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: adc.kubeClient.CoreV1().Events("")})
 	defer adc.broadcaster.Shutdown()
 
-	klog.Infof("Starting attach detach controller")
-	defer klog.Infof("Shutting down attach detach controller")
+	logger := klog.FromContext(ctx)
```

Comment:
I don't like passing a logger wrapped in `ctx` and then unpacking it here. Can `NewAttachDetachController` just store it and then use it as `adc.logger`? What value has passing the logger as a parameter everywhere?

The same applies to other objects, like the populator, ActualStateOfWorld and DesiredStateOfWord and other controllers in general.

Reply:
Ack, let's pass it as argument then.